R/additional_indicators.R

```diff
@@ -0,0 +1,129 @@
+#' Add set of extra indicators to pip output
+#'
+#' @param dt data.frame: country level out from PIP
+#'
+#' @return data.frame
+#' @noRd
+additional_indicators <- function(dt) {
+
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  # ADd variables   ---------
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ## get current var names --------
+  # df <- as.data.frame(dt[1, ])
+  original_names <- copy(names(dt))
+
+  #~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ## Add new navariables --------
+
+  dt[,
+      bottom40 := decile1 + decile2 + decile3 + decile4,
+  ][,
+    `:=`(
+      pop_in_poverty = wbpip::get_lh_number_poor(
+           headcount = headcount,
+           pop       = reporting_pop),
+
+      average_shortfall = wbpip::get_lh_average_shortfall(
+        headcount = headcount,
+        povgap    = poverty_gap,
+        povline   = poverty_line),
+
+      total_shortfall = wbpip::get_lh_total_shortfall(
+        headcount = headcount,
+        pop       = reporting_pop,
+        povgap    = poverty_gap,
+        povline   = poverty_line),
+
+      income_gap_ratio = wbpip::get_lh_income_gap_ratio(
+        headcount = headcount,
+        povgap    = poverty_gap),
+
+      palma_ratio = wbpip::get_lh_palma_ratio(
+        top10     = decile10,
+        bottom40  = bottom40),
+
+      p90p10_ratio = wbpip::get_lh_9010_ratio(top10    = decile9,
+                                              bottom10 = decile1)
+
+    )
+  ]
```

Comment:
In my experience, using Non-Standard Evaluation is always slower than using standard evaluation.
I think something like this will most likely be faster:
 `dt$pop_in_poverty <-  wbpip::get_lh_number_poor(headcount = dt$headcount, pop = dt$reporting_pop)`

But please don't take my word for it, and use the `bench` or `microbenchmark` to decide on the fastest implementation.

Reply:
Hi @tonyfujs , 

I [tested](https://gist.github.com/randrescastaneda/9583229d58818cea5f2db642f184cf84) the performance and found out that NSE is way faster than SE. I think the reason is that each time we create a variable in the data.frame the data is copy in memory, which consumes a lot of time. `data.table` is super efficient in this kind of calculations. I think we are good leaving it like that. 

![image](https://github.com/PIP-Technical-Team/pipapi/assets/35301997/d9203891-635e-4d44-acf4-839c30ded38f)

Reply:
Uuuf... Way faster indeed! Sounds good then